workflow/artifacts/git/git.go

```diff
@@ -96,7 +111,7 @@ func (g *GitArtifactDriver) Save(string, *wfv1.Artifact) error {
 }
 
 func (g *GitArtifactDriver) Load(inputArtifact *wfv1.Artifact, path string) error {
-	closer, auth, env, err := g.auth()
+	closer, auth, env, err := g.auth(inputArtifact.Git.Repo)
```

Comment:
could pass the username here instead of whole URL

Reply:
updated to pass username in stead of repo url